packages/suite/src/actions/suite/analyticsActions.ts

```diff
@@ -311,15 +334,27 @@ export const init = (loadedState: State, optout: boolean) => (
         payload: {
             instanceId,
             sessionId,
+            sessionStart: Date.now(),
             enabled,
         },
     });
     // 5. if analytics was initiated as enabled, continue with setting up side effects
     if (!getState().analytics.enabled) return;
     // 6. error logging to sentry
     setSentryUser(instanceId);
-    // 7. register event listeners to report events from electron
-    // todo:
+    // 7. register event listeners
+    // @ts-ignore. how do we handle native with respect to window specific event listeners?
+    window.addEventListener('beforeunload', () => {
```

Comment:
move it to utils

Reply:
Moved it into utils, but still I don't feel 100% happy. Maybe there could be a better, more generic solution. But I would need to dive into react-native a bit first to find out how these listners are used there. If there is one global object, a counter part to window, this function could be like `setGlobalListener(event, callback)`